deploy/crds/nodemaintenance_crd.yaml

```diff
@@ -17,23 +16,51 @@ spec:
     openAPIV3Schema:
       properties:
         apiVersion:
+          description: 'APIVersion defines the versioned schema of this representation
+            of an object. Servers should convert recognized schemas to the latest
+            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources'
           type: string
         kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
           type: string
         metadata:
           type: object
         spec:
           properties:
             nodeName:
+              description: Node name to apply maintanance on/off
               type: string
             reason:
+              description: Reason for maintanance
               type: string
           required:
           - nodeName
           type: object
         status:
           properties:
-            phase:
+            completed:
+              description: Completed represents the state of maintenance. True if
+                completed , false otherwise
+              type: boolean
+            errorMessage:
```

Comment:
Could we use Conditions to indicate specific error conditions along with human readable string?

Reply:
Well there was a debate whether we should still use conditions or not and i refrained from using condition till now.  how critical is it to use conditions instead of a error string field that would indicate the last error ?

Reply:
probably not a big deal